### PR TITLE
fix: fix #260 UPDATE with LIMIT and ORDER should be formatted(mysql only)

### LINKED ISSUE #260: [bug] UPDATE with LIMIT and ORDER should be formatted
```SQL
// mysql only
UPDATE users SET updated_at = "1977-01-01 19:22:22" ORDER BY id LIMIT 4
```

### PR DESCRIPTION
**Pull request by JimmyDaddy**
attr

### CODE REVIEW

File: src/drivers/abstract/spellbook.js [hunk 1]
```diff
@@ -476,12 +476,22 @@ function formatUpdate(spell) {
     for (const condition of whereConditions) collectLiteral(spell, condition, values);
     chunks.push(`WHERE ${formatConditions(spell, whereConditions)}`);
   }
+
+  const extraOptions = this.formatUpdateExtraOptions(spell);
+  if (extraOptions.length) {
+    chunks.push(...extraOptions);
+  }
```
Comment (cyjake) [inline src/drivers/abstract/spellbook.js:483]: This bug fix shouldn't be sequelize adapter specific I think. Use cases like below should be supported as well:

```js
Post.update({ title: null }, { title: 'Untitled' }).order('id', 'desc').limit(10);
```

Comment (JimmyDaddy) [reply src/drivers/abstract/spellbook.js:483]: It a MySQL only feature, this was supported at this PR, see 
* test/integration/suite/basics.test.js:685
* test/unit/spell.test.js:638

Comment (JimmyDaddy) [reply src/drivers/abstract/spellbook.js:483]: > This bug fix shouldn't be sequelize adapter specific I think. Use cases like below should be supported as well:
> 
> ```js
> Post.update({ title: null }, { title: 'Untitled' }).order('id', 'desc').limit(10);
> ```
It's a MySQL only feature, this was supported at this PR, see

test/integration/suite/basics.test.js:685
test/unit/spell.test.js:638

File: test/integration/suite/basics.test.js [hunk 2]
```diff
@@ -682,6 +682,21 @@ describe('=> Basic', () => {
       expect(posts.length).to.equal(affectedRows);
     });
 
+    it('Bone.update(where, values) should work with limit/offset and order', async function() {
+      if (Post.driver.type === 'mysql') {
```
Comment (JimmyDaddy) [inline test/integration/suite/basics.test.js:686]: mysql only

**Review by cyjake — APPROVED** [review]

**Pull request merged**